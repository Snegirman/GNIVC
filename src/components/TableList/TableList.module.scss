.table {
  text-align: center;
  padding: 10px;
  width: 100%;
}

.thead {
  font-size: 20px;

  .tableCell {
    width: 40vw;
    border-top: 3px solid #aaaaaa;
    border-bottom: 3px solid #aaaaaa;
    border-right: none;
  }
}

.tableCell {
  padding: 10px;
  width: 40vw;
  border-bottom: 1px solid #aaaaaa;
}

.tableRow {
  height: 8vh;
  width: 100%;
}


.tableLineBtnContainer {
  display: flex;
  justify-content: space-around;
  font-size: 12px;

  @media screen and (min-width: 1024px) {
    justify-content: center;
    padding: 10px;
  }
  
  .tableBtn {
    background-color: white;
    border: none;
    font-size: 12px;
  }

  .goToPost {
    text-decoration: none;
    color: black;
    background-color: white;
    padding: 5px;
    margin-left: 10px;
  }
}
