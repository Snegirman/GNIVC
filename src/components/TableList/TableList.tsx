import _ from 'lodash';
import React, { Dispatch, FC, SetStateAction } from 'react';

import { Post as PostType } from '../../models/post';
import Loading from '../Loading';
import Error from '../Error';
import classes from './TableList.module.scss';

type TableListProps = {
  children: React.ReactNode;
  postData: PostType[];
  setSortedData: Dispatch<SetStateAction<PostType[]>>;
  isError: boolean;
  isLoading: boolean;
};

const TableList: FC<TableListProps> = (props) => {
  const {
    children, postData, setSortedData, isLoading, isError,
  } = props;

  const sortCategoryHandler = () => {
    setSortedData(_.sortBy(postData, (o) => o.category));
  };

  const sortJokeHandler = () => {
    setSortedData(_.sortBy(postData, (o) => o.joke));
  };

  if (isLoading) {
    return <Loading />;
  }

  if (isError) {
    return <Error />;
  }

  return (
    <table className={classes.table} cellSpacing="0">
      <thead className={classes.thead}>
        <tr className={classes.tableRow}>
          <th className={classes.tableCell} onClick={sortCategoryHandler}>
            Category
          </th>
          <th className={classes.tableCell} onClick={sortJokeHandler}>
            Joke
          </th>
        </tr>
      </thead>
      <tbody>
        {children}
      </tbody>
    </table>
  );
};

export default TableList;
