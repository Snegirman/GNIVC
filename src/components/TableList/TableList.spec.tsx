import React from 'react';
import { render, screen } from '@testing-library/react';

import TableList from './index';

const post = {
  category: 'category',
  flags: {
    explicit: false,
    nsfw: false,
    political: false,
    racist: false,
    religious: false,
    sexist: false,
  },
  id: 1,
  joke: 'some joke',
  lang: 'en',
  safe: true,
  type: 'any',
};

describe('Component table line', () => {
  it('Should render', () => {
    render(
      <TableList
        postData={[post]}
        setSortedData={() => { }}
        isError={false}
        isLoading={false}
      ><tr><td>Some text</td></tr>
      </TableList>,
    );
    expect(screen.getByText('Some text')).toBeInTheDocument();
  });

  it("Should didn't change snapshot", () => {
    const app = render(
      <TableList
        postData={[post]}
        setSortedData={() => { }}
        isError={false}
        isLoading={false}
      ><tr><td>Some text</td></tr>
      </TableList>,
    );
    expect(app).toMatchSnapshot();
  });
});
