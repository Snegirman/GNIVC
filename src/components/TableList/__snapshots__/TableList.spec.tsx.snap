// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component table line Should didn't change snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        cellspacing="0"
        class="table"
      >
        <thead
          class="thead"
        >
          <tr
            class="tableRow"
          >
            <th
              class="tableCell"
            >
              Category
            </th>
            <th
              class="tableCell"
            >
              Joke
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Some text
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      cellspacing="0"
      class="table"
    >
      <thead
        class="thead"
      >
        <tr
          class="tableRow"
        >
          <th
            class="tableCell"
          >
            Category
          </th>
          <th
            class="tableCell"
          >
            Joke
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Some text
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
